src/commands/sourcemaps/git.ts

```diff
@@ -0,0 +1,208 @@
+    maxConcurrentProcesses: 3,
+  }
+
+  return simpleGit.gitP(options)
+}
+
+// GitRemote returns the remote of the current repository.
+export const gitRemote = async (git: simpleGit.SimpleGit): Promise<string> => {
+  const remotes = await git.getRemotes(true)
+  if (remotes.length === 0) {
+    throw new Error('No git remotes available')
+  }
+  remotes.forEach((remote) => {
+    if (remote.name === 'origin') {
+      return remote.refs.push
+    }
+  })
+
+  return remotes[0].refs.push
+}
+
+// StripCredentials removes credentials from a remote HTTP url.
+export const stripCredentials = (remote: string) => {
+  try {
+    const url = new URL(remote)
+    url.username = ''
+    url.password = ''
+
+    return url.toString()
+  } catch {
+    return remote
+  }
+}
+
+// GitHash returns the hash of the current repository.
+export const gitHash = async (git: simpleGit.SimpleGit): Promise<string> => git.revparse('HEAD')
+
+// GitTrackedFiles returns the tracked files of the current repository.
+export const gitTrackedFiles = async (git: simpleGit.SimpleGit): Promise<string[]> => {
+  const files = await git.raw('ls-files')
+
+  return files.split(/\r\n|\r|\n/)
+}
+
+// TrimStart trims from a set of characters from the start of a string.
+export const trimStart = (str: string, chars: string[]) => {
+  let start = 0
+  const end = str.length
+  while (start < end && chars.indexOf(str[start]) >= 0) {
+    ++start
+  }
+
+  return start > 0 ? str.substring(start, end) : str
+}
+
+// Trim trims from a set of characters from a string.
+export const trim = (str: string, chars: string[]) => {
+  let start = 0
+  let end = str.length
+  while (start < end && chars.indexOf(str[start]) >= 0) {
+    ++start
+  }
+  while (end > start && chars.indexOf(str[end - 1]) >= 0) {
+    --end
+  }
+
+  return start > 0 || end < str.length ? str.substring(start, end) : str
+}
+
+// CleanupSource generates a proper source file path from a sourcemap:
+// - Strip a set of hard-coded prefixes ('webpack:///./')
+// - Strip the eventual projectPath
+// - Removes query parameters
+//
+// It returns the new source as well as wether are not the specified projectPath was stripped.
+export const cleanupSource = (source: string, projectPath: string): [string, boolean] => {
+  // Prefixes
+  const prefixesToRemove = ['webpack:']
+  for (const p of prefixesToRemove) {
+    if (source.startsWith(p)) {
+      source = source.slice(p.length)
+    }
+  }
+  source = trimStart(source, ['/', '.'])
+  // ProjectPath
+  projectPath = trim(projectPath, ['/', '.'])
+  const projectPathFound = source.substr(0, projectPath.length) === projectPath
+  if (projectPathFound) {
+    source = source.slice(projectPath.length)
+  }
+  // Query parmeter
+  const pos = source.lastIndexOf('?')
+  if (pos > 0) {
+    source = source.slice(0, pos)
+  }
+
+  return [trimStart(source, ['/', '.']), projectPathFound]
+}
+
+// TrackedFilesMap transforms a list of tracked files into a map to look up sources.
```

Comment:
Maybe we could add a bit more details to the docstring? More generally speaking I think the algorithm used to do the matching could be more documented.

Reply:
I added several comments explaining what the functions do and how to use them. Tell me if it makes more sense.